internal/security/secret.go

```diff
@@ -28,8 +28,10 @@ import (
 
 // SecretProvider cache storage for the secrets
 type SecretProvider struct {
-	secretClient pkg.SecretClient
-	secrets      map[string]map[string]string
+	secretClient  pkg.SecretClient
+	secrets       map[string]map[string]string // secret's path, key, value
+	configuration *common.ConfigurationStruct
```

Comment:
I would save a pointer to just the `InsecureSecrets`, not the whole config struct. Needs to be a pointer so that if the values change they are reflected here.